addons/test_mail/tests/test_mail_lead.py

```diff
@@ -0,0 +1,58 @@
+from datetime import datetime, timezone, timedelta
+from odoo.addons.mail.tests.common import MailCommon
+from odoo.tests import tagged
+
+
+@tagged("post_install", "-at_install")
+class TestCrmLeadActivityState(MailCommon):
+    def test_groupby_activity_state_progress_bar_behavior(self):
+        """Test activity_state groupby logic on mail.test.lead"""
+
+        def verify_activity(res_id, expected_state, days=0):
+            activity = self.env["mail.activity"].create({
+                "summary": f"Test activity for CRM lead {id}",
+                "res_model_id": self.env["ir.model"]._get_id("mail.test.lead"),
+                "res_id": res_id,
+                "date_deadline": datetime.now(timezone.utc) + timedelta(days=days),
+                "user_id": self.env.user.id,
+            })
+            self.assertEqual(activity.state, expected_state)
```

Comment:
Why not but this should already be mostly tested, I think the check on the progress bar result is sufficient